R/pkg/R/serialize.R

```diff
@@ -37,24 +37,38 @@ writeObject <- function(con, object, writeType = TRUE) {
   # passing in vectors as arrays and instead require arrays to be passed
   # as lists.
   type <- class(object)[[1]]  # class of POSIXlt is c("POSIXlt", "POSIXt")
-  if (writeType) {
-    writeType(con, type)
+  isNA <- FALSE
+  if (type %in% c("integer", "character", "logical", "double", "numeric")) {
```

Comment:
Is there a reason why we need this `type %in%` check ? Could you add a comment on top with it ?